### PR TITLE
refactor(deps): using local difference instead of lodash

### PR DESCRIPTION
**Pull request by sarbbottam**
@ta2edchimp  Do we need `lodash.difference` for our use-case?

### CODE REVIEW

File: test/difference.js [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import test from 'ava'
+
+import differnce from '../src/difference'
```
Comment (ta2edchimp) [inline test/difference.js]: There's a typo (differ**_e_**nce) ...

File: src/difference.js [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+    hash[item] = true
+  })
+
+  a.forEach(function findDiff(item) {
+    if (!hash[item] && !diff[item]) {
+      diff[item] = true
+    }
+  })
+
+  return Object.keys(diff)
```
Comment (ta2edchimp) [inline src/difference.js]: I don't remember that `Object.keys` or `Object.prototype.forEach` are part of v8's language implementations in node `0.10` (or `0.12`). It could possibly just work within the tests' environment because they're transpiled with babel...

`Object.property.forEach` even isn't spec, or is it?

**Comment by ta2edchimp** [comment]
I'm unsure about this. On one hand, I often don't see the benefit of requiring modules (and the overhead they bring in) for one-liner functionality. On the other hand (if it's not only those "one-liners"), I'd like to keep everything DRY — or at least not reinvent the wheel.

I'm really unsure, but I'd follow your impression of the need to get rid of this dependency, if we sort out the language implementation issue I noted about `Object.keys` and `Object.prototype.forEach`.

File: src/difference.js (hunk 2 above)
Comment (ta2edchimp) [inline src/difference.js]: Aaaand now I realized it's of course `Array.prototype.forEach`. My bad — I was caught by my suspicion, `Object.keys` wouldn't have been available in node `0.1x`.

So, I'm good with these lines 😉

**Comment by ta2edchimp** [comment]
**tl;dr:** If you think it would be a good thing to get rid of this dependency, I'd say merge it (after fixing that typo).

**Comment by sarbbottam** [comment]
Thanks for detailed review, much appreciated. `Array.prototype.forEach` & `Object.keys`, and it works, I was quite confident, as at my work we are still in node 10.X and using ES5 (no babel), however still verified it. 
```js
吽 node
> var diff = require('./src/difference');
undefined
> diff(['a', 'b', 'c'], ['x', 'y', 'z']);
[ 'a', 'b', 'c' ]
> process.version
'v0.10.43'
>
```

I would fix the typo and rebase the PR.

**Comment by sarbbottam** [comment]
>I'd say merge it (after fixing that typo).

Merging it on @ta2edchimp's behalf.

**Comment by ta2edchimp** [comment]
Yeah, I verified it too (and re the `forEach`: I was completely distracted by my anger vs `npm@2.x` :stuck_out_tongue_winking_eye: )

Pull the trigger! :tada:

**Comment by ta2edchimp** [comment]
Epic synchronous posting :stuck_out_tongue_winking_eye:
